### ISSUE TITLE
Fake Waterlogging

### ISSUE THREAD

**Issue by StrideyGH** [sp614x/optifine#1580]
There are still a lot of blocks that can't be water logged in 1.13. Much like the fake snow layer under the fence feature, I suggest that blocks surrounded by water (that can't already be waterlogged) get fake waterlogged (water texture around the block). Obviously you'd still be able to breathe in these spots but at least it would be nice aesthetically.
**Blocks that aren't waterloggable:** 
- Cake
- Turtle eggs
- Leaves
- Chorus flowers (dead and live)
- Farmland (wet and dry)
- Enchanting table
- End portal frame
- End portal
- End gateway
- Anvil 
- Grass path
- Banners (floor and wall)
- Fence gates
- Daylight sensors
- Droppers
- Doors (all wood and iron)
- Dragon egg
-  Beds
- Monster spawners
- Pressure plates
- Barriers
- Sugar cane

https://www.reddit.com/r/Minecraft/comments/904lw7/blocks_that_still_cant_be_watterlogged_by_default/

**Comment by Regnander** [comment]
Not to be a mood killer, but I see two flaws in this.

### Confusion
This could be very confusing to distinguish real waterlogged blocks with the fake ones.

>"I'm gonna jump down into that waterlogged sugar cane so I don't take any fall damage"
>[jumps and dies of fall damage]
>"Why didn't the water break my fall?! Did the game gli- Oh, right, OptiFine."
### Implementation
If a player creates a waterlogged dropper and another player decides to log in and approach the dropper on a server, how could the client of the latter know that the specific dropper should be waterlogged?

**Comment by realjackasterisk** [comment]
> "I'm gonna jump down into that waterlogged sugar cane"  

That sounds like an *incredibly* specific and unlikely situation.

Also, I think "dropper" was supposed to be "hopper", but anyway..
The idea is to waterlog blocks that are already surrounded by water, similar to Better Snow. So in this case, to create a waterlogged hopper, the hopper would have to be placed directly next to water.

**Comment by realjackasterisk** [comment]
I'm surprised that wasn't implemented in 1.13 to begin with.

**Comment by nfitzen** [comment]
A Cauldron is basically able to be waterlogged because you can fill it up.

As for the others, I think water should have to be placed **all around it** as in, water should be on top as well, to prevent anybody from being confused.

Also, that option should be disabled by default so that nobody gets confused when they can breathe in "waterlogged" sugarcane.

**Comment by ontvlambaargames** [comment]
Err, sorry to ruin your input, but certain blocks will be destroyed by water, unless the water is falling. 

Sugarcane seems to be a hot topic here. Having a water source next to the sugarcane will definitely destroy the sugarcane. Waterlogged sugarcane requires the user to place the four sources above the highest sugarcane and needs to drop into a four spaces around the sand/gravel/dirt/etc. the sugarcane is planted on.

Chorus flowers are grown by chorus plants, which can also be broken by water. 
Farmland and the grass path will eventually turn into dirt, when there's water over it for too long. The only possibility here seems to place them next to a water source and render the water over that missing one pixel in those blocks. It'll be impossible to use the blocks underwater.

**Comment by Madis0** [comment]
I'm mostly against the idea for reasons @Regnander listed, but I could see it being useful for decoration in singleplayer, therefore I would suggest two changes if this gets implemented:

- add it only for blocks where a player cannot reasonably breathe in (e.g. _not_ doors and fence gates, maybe more)
- make it off by default

**Comment by nfitzen** [comment]
@ontvlambaargames yeah, [Sugar cane](//minecraft.gamepedia.com/Sugar_Cane) will be destroyed. I was only using it as an example (a bad one at that). However, I do think [Barriers](//minecraft.gamepedia.com/Barrier) and [Leaves](//minecraft.gamepedia.com/Leaves), for example, would be great for this idea.

I definitely agree with @Madis0's point that it should only be for blocks that one cannot reasonably breathe in.

Also, why didn't [Mojang](//www.mojang.com/) add this in the first place?
